packages/transaction-manager/test/index.test.ts

```diff
@@ -339,6 +355,165 @@ describe('index', () => {
+          TestData.idRaw3.encryptionParams,
+        ]);
+
+        expect(ret.result).toEqual({});
+        expect(ret.meta).toEqual({
+          dataAccessMeta: { persisted: false },
+          encryptionMethod: 'ecies-aes256-gcm',
+        });
+        expect(ret).toHaveProperty('inMemoryTransaction');
+        expect(fakeDataAccess.persistTransaction).not.toHaveBeenCalled();
+      });
+    });
+
+    describe('in an existing new channel', () => {
+      afterEach(() => {
+        jest.clearAllMocks();
+      });
+      it('can persist a clear transaction in an existing channel', async () => {
+        const transactionManager = new InMemoryTransactionManager(fakeDataAccess);
+
+        const ret = await transactionManager.persistTransaction(data2, channelId, extraTopics);
+
+        expect(ret.result).toEqual({});
+        // 'ret.meta is wrong'
+        expect(ret.meta).toEqual({
+          dataAccessMeta: { persisted: false },
+          encryptionMethod: undefined,
+        });
+        expect(ret).toHaveProperty('inMemoryTransaction');
+        expect(fakeDataAccess.persistTransaction).not.toHaveBeenCalled();
+      });
+
+      it('can persist a encrypted transaction in an existing channel', async () => {
+        const encryptedTx = await TransactionsFactory.createEncryptedTransactionInNewChannel(data, [
+          TestData.idRaw1.encryptionParams,
+        ]);
+
+        const fakeMetaDataAccessGetReturnWithEncryptedTransaction: DataAccessTypes.IReturnGetTransactions =
+          {
+            meta: {
+              transactionsStorageLocation: ['fakeDataId1'],
+            },
+            result: {
+              transactions: [
+                {
+                  state: TransactionTypes.TransactionState.PENDING,
+                  timestamp: 1,
+                  transaction: encryptedTx,
+                },
+              ],
+            },
+          };
+
+        fakeDataAccess = {
+          _getStatus: jest.fn(),
+          getChannelsByMultipleTopics: jest.fn(),
+          getChannelsByTopic: jest.fn(),
+          getTransactionsByChannelId: jest
+            .fn()
+            .mockReturnValue(fakeMetaDataAccessGetReturnWithEncryptedTransaction),
+          initialize: jest.fn(),
+          close: jest.fn(),
+          persistTransaction: jest.fn().mockReturnValue(fakeMetaDataAccessPersistReturn),
+        };
+
+        const transactionManager = new InMemoryTransactionManager(
+          fakeDataAccess,
+          TestData.fakeDecryptionProvider,
+        );
+        const ret = await transactionManager.persistTransaction(data2, channelId, extraTopics);
+
+        expect(ret.result).toEqual({});
+        expect(ret.meta).toEqual({
+          dataAccessMeta: { persisted: false },
+          encryptionMethod: 'ecies-aes256-gcm',
+        });
+
+        expect(fakeDataAccess.persistTransaction).not.toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe('persistInMemoryTransaction', () => {
```

Comment:
Did not test error handling an new/existing channel behavior since it's already tested in the regular `TransactionManager` since they both use the same helper functon.